src/legends/swatches.js

```diff
@@ -0,0 +1,71 @@
+import {html} from "htl";
+import uid from "../uid.js";
+
+export function swatches(color, {
+  columns = null,
+  format = x => x,
+  swatchSize = 15,
+  swatchWidth = swatchSize,
+  swatchHeight = swatchSize,
+  marginLeft = 0
+} = {}) {
+  const id = uid().id;
+
+  if (columns !== null) return html`<div style="display: flex; align-items: center; margin-left: ${+marginLeft}px; min-height: 33px; font: 10px sans-serif;">
+  <style>
+
+.${id}-item {
+  break-inside: avoid;
+  display: flex;
+  align-items: center;
+  padding-bottom: 1px;
+}
+
+.${id}-label {
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  max-width: calc(100% - ${+swatchWidth}px - 0.5em);
+}
+
+.${id}-swatch {
+  width: ${+swatchWidth}px;
+  height: ${+swatchHeight}px;
+  margin: 0 0.5em 0 0;
+}
+
+  </style>
```

Comment:
We can move all of this to style.css and then just use CSS variables for the variables (the swatch width, height, and color). Then we won’t need uid or inline styles (other than to define the CSS variables), either.